torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -813,19 +833,35 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
   }
 
   if (execute_kernel_) {
-    FUSER_PERF_SCOPE("ExecutorRunFusion::cuLaunchKernel");
-    AT_CUDA_DRIVER_CHECK(at::globalContext().getNVRTC().cuLaunchKernel(
-        compiled_kernel_.function,
-        launch_params.gdimx(),
-        launch_params.gdimy(),
-        launch_params.gdimz(),
-        launch_params.bdimx(),
-        launch_params.bdimy(),
-        launch_params.bdimz(),
-        launch_params.smem(),
-        stream,
-        kernel_arguments.getBuffer(),
-        nullptr));
+    if (!kernel()->summary().has_cooperative_grid_reduction) {
+      FUSER_PERF_SCOPE("ExecutorRunFusion::cuLaunchKernel");
+      AT_CUDA_DRIVER_CHECK(at::globalContext().getNVRTC().cuLaunchKernel(
+          compiled_kernel_.function,
+          launch_params.gdimx(),
+          launch_params.gdimy(),
+          launch_params.gdimz(),
+          launch_params.bdimx(),
+          launch_params.bdimy(),
+          launch_params.bdimz(),
+          launch_params.smem(),
+          stream,
+          kernel_arguments.getBuffer(),
+          nullptr));
+    } else {
+      FUSER_PERF_SCOPE("ExecutorRunFusion::cuLaunchCooperativeKernel");
+      AT_CUDA_DRIVER_CHECK(
+          at::globalContext().getNVRTC().cuLaunchCooperativeKernel(
```

Comment:
nvfuser is also enabled for rocm. cuLaunchCooperativeKernel is only added for cuda, so we need to guard the symbol here with `#ifndef __HIP_PLATFORM_HCC__` and assert on the else block.